packages/botonic-cli/src/analytics/credentials-handler.ts

```diff
@@ -0,0 +1,123 @@
+import { homedir } from 'os'
+import { join, resolve } from 'path'
+import { v4 as uuidv4 } from 'uuid'
+
+import {
+  BOT_CREDS_FILENAME,
+  BOTONIC_HOME_DIRNAME,
+  BOTONIC_PROJECT_PATH,
+  GLOBAL_CREDS_FILENAME,
+} from '../constants'
+import {
+  BotCredentials,
+  CredentialsHandlerArgs,
+  GlobalCredentials,
+} from '../interfaces'
+import { create, pathExists, readJSON, writeJSON } from '../util/file-system'
+import { execCommand, isWindows } from '../util/processes'
+
+export class CredentialsHandler {
+  homePath: string
+  pathToCredentials: string
+
+  constructor(args: CredentialsHandlerArgs) {
+    this.homePath = resolve(args.homePath)
+    this.pathToCredentials = join(this.homePath, args.filename)
+    this.initialize()
+  }
+
+  initialize(): void {
+    this.createIfNotExists()
+  }
+
+  generateId(): string {
+    return uuidv4()
+  }
+
+  createIfNotExists(): void {
+    if (!pathExists(this.homePath)) create(this.homePath)
+  }
+
+  load(): any {
+    try {
+      return (
+        pathExists(this.pathToCredentials) && readJSON(this.pathToCredentials)
+      )
+    } catch (e) {
+      console.warn('Credentials could not be loaded')
```

Comment:
I decided not to print this exception because the developer flow continues working as expected despite of not being available of loading the credentials and don't want to show a large error within the flow (the handicap here, and this fail will be due to user permissions is that for every deploy, the user must login again).

Reply:
I understand, but hiding exceptions is a no-no. I suggest implementing here  a simple function which only displays the error message but not the call stack, and only do it once per botonic execution to minimize spamming. I think you implemented something similar in botonic, but here it can be much simpler as a start.
```
let credentialsFailed = false
function logCredentialsFailure(msg: string, e) {
  if (!credentialsFailed) {
     console.warng(msg +  ":" + str(e)
    credentialsFailed = true
}
}

```